plaid/client.py

```diff
@@ -455,3 +456,28 @@ def institution(self, url, institution_id):
             url.format(institution_id),
             suppress_errors=self.suppress_http_errors
         )
+
+    @inject_url('institution_search')
+    def institution_search(self, url, q=None, p=None, institution_id=None):
+        '''
+        Perform simple search query against Long Tail institutions.
+
+        `q`   str    Query to match against the full list of institutions.
+        `p`    str   Filter FIs by a single product (Optional).
+        `institution_id`  str
+                    The id of a single institution for lookup (Optional).
+        '''
+
+        assert q is not None or institution_id is not None, (
+            'query or institution_id required'
+        )
+
+        params = dict([(key, value) for key, value in [
+            ('q', q),
+            ('p', p),
+            ('id', institution_id)
+        ] if value is not None])
```

Comment:
If only there were a functional programming library in Python which supported filtering dictionaries. ;)